src/test/java/com/google/research/bleth/simulator/ObserverTest.java

```diff
@@ -0,0 +1,930 @@
+    private static final Location oneOnZeroCoordinate = new Location(1, 0);
+    private static final BeaconFactory beaconFactory = new BeaconFactory();
+    private static final ObserverFactory observerFactory = new ObserverFactory();
+
+    @Mock
+    private Simulation simulation;
+
+    /**
+     * A fake implementation for IResolver that allows to test the arguments it receives when an observer calls receiveInformation.
+     */
+    private class FakeResolver implements IResolver {
+        private Location observerLocation;
+        private List<Transmission> transmissions;
+
+        @Override
+        public void receiveInformation(Location observerLocation, List<Transmission> transmissions) {
+            this.observerLocation = observerLocation;
+            this.transmissions = transmissions;
+        }
+
+        public Location getObserverLocation() {
+            return observerLocation;
+        }
+
+        public List<Transmission> getTransmissions() {
+            return transmissions;
+        }
+
+        @Override
+        public Board getBoard() {
+            return null;
+        }
+    }
+
+    private IResolver resolver = new FakeResolver();
+
+    private Observer createStaticObserverOnLocation(Location initial_location) {
+        return observerFactory.createObserver(initial_location, new StationaryMovementStrategy(), resolver, simulation,
+        1, 1, new FixedAwakenessStrategy());
+    }
+
+    private Observer createRandomObserverOnLocation(Location initial_location) {
+        return observerFactory.createObserver(initial_location, new RandomMovementStrategy(), resolver, simulation,
+                1, 1, new FixedAwakenessStrategy());
+    }
+
+    private Observer createObserverByAwakenessTimes(int awakenessDuration, int firstAwakenessTime, AwakenessStrategy awakenessStrategy) {
+        return observerFactory.createObserver(zeroOnZeroCoordinate, new RandomMovementStrategy(), resolver, simulation,
+                awakenessDuration, firstAwakenessTime, awakenessStrategy);
+    }
+
+    private Beacon createStaticBeaconOnLocation(Location initial_location) {
+        return beaconFactory.createBeacon(initial_location, new StationaryMovementStrategy(), simulation);
+    }
+
+    private int calculateDistance(Location oldLocation, Location newLocation) {
+        return Math.abs(newLocation.row - oldLocation.row) + Math.abs(newLocation.col - oldLocation.col);
+    }
+
+    private boolean isMovementOfCorneredObserverIsValid(Board board, Location corner) {
+        Mockito.when(simulation.getBoard()).thenReturn(board);
+        Observer randomObserver = createRandomObserverOnLocation(corner);
+
+        Location nextLocation = randomObserver.moveTo();
+
+        return simulation.getBoard().isLocationValid(nextLocation) && calculateDistance(corner, nextLocation) <= 1;
+    }
+
+    private boolean movingObserverLocationIsUpdatedOnBoardMatrix(Board board, Location initialLocation) {
```

Comment:
This is a helper, but it has enough logic (creating objects/modifying/asserting) that it seems it should just be a test on its own.